src/scenechange/mod.rs

```diff
@@ -7,43 +7,25 @@
 // Media Patent License 1.0 was not distributed with this source code in the
 // PATENTS file, you can obtain it at www.aomedia.org/license/patent.
 
-use crate::api::EncoderConfig;
 use crate::api::InterConfig;
-use crate::frame::Frame;
-use crate::util::CastFromPrimitive;
+use crate::api::{EncoderConfig, FrameType};
 use crate::util::Pixel;
 
+use crate::encoder::FrameInvariants;
 use std::cmp;
 use std::collections::BTreeSet;
-use std::sync::Arc;
+
+const SCENECUT_THRESHOLD: f32 = 40.;
```

Comment:
Where did this come from?

Reply:
x264. I did some testing as well to see if this is really the ideal threshold and I wasn't able to come to a solid conclusion. But the way it scales seems reasonable, because quality degradation gets worse the further you get from the last keyframe. So, this is the max threshold if you're near the max keyframe interval. Near the min interval, it's divided by 4, so it ends up starting at .1. Note also that the ratios can go negative (if inter cost is higher than intra cost), but we don't always want to wait for that to signal a new keyframe (on some sources, it may be very rare for inter cost to be greater than intra cost even on what a human would perceive as a "scene change").